notes/stv/20191209-openstack-deployment.txt

```diff
@@ -0,0 +1,919 @@
+################################################################
+
+
+# Read up and try to follow
+# https://docs.hpc.cam.ac.uk/cloud-training/#16
+
+
+#-------------------------------------------------------------------------
+# GUI Steps to Setup Networks, Security Groups & Instances
+#-------------------------------------------------------------------------
+
+
+## Step 1:
+
+Create Private Network 
+
+   Name: stv-dev-network
+
+   Create Subnet
+      Name - stv-dev-subnet 
+      CIDR - 10.0.0.0/24
+      DNS Name Servers - 131.111.8.42 131.111.12.20      
+
+
+
+## Step 2:
+
+Create Router, connected to Internet
+ 
+   Name: stv-dev-router
+   External Network: Internet
+
+   Create Interface
+      Subnet - stv-dev-network
+
+
+## Step 3
+
+Create Security Groups (based on training presentation)
+
+external-bastion: 22 / ICMP
+internal-bastion: default
+internal-webserver: 22 - internal-bastion
+external-webserver: 80
+stv-dev-zeppelin : 8080
+stv-dev-master: *
+stv-dev-worker: * - stv-dev-master
+
+
+
+## Step 4:
+
+Create Bastion Instance (Gateway)
+    Fedora-30-1.2
+    general.v1.tiny	
+    external-bastion + internal-bastion
+
+
+## Step 5:
+
+Create Master node
+    Fedora-30-1.2
+    general.v1.medium	
+    internal-bastion + internal-webserver + stv-dev-master
+
+
+
+## Step 6:
+
+Create 6 Worker Nodes with 500 Gb volumes
+    Fedora-30-1.2
+    general.v1.large	
+    (external-bastion + internal-webserver + stv-dev-worker)
+
+
+
+## Step 7:
+
+Create Zeppelin Node 
+(external-bastion + internal-webserver + stv-dev-zeppelin)
+
+
+
+
+
+stv-dev-gateway-ip=
+stv-dev-worker-1-ip=
+stv-dev-worker-2-ip=
+stv-dev-worker-3-ip=
+stv-dev-master-ip=
+stv-dev-zeppelin-ip=
+
+
+
+
+#-------------------------------------------------------------------------
+# Setup .ssh config locally
+#-------------------------------------------------------------------------
+
+~/.ssh/config
+..
+
+
+Host stv-dev-gateway
+    User centos
+    IdentityFile ~/.ssh/stv-master
+    Hostname ${stv-dev-gateway-ip:?}
+
+Host stv-dev-master
+    User fedora
+    IdentityFile ~/.ssh/stv-master
+    Hostname${stv-dev-master-ip:?}
+    ProxyCommand ssh -W %h:%p stv-dev-gateway
+
+Host stv-dev-worker-1
+    User fedora
+    IdentityFile ~/.ssh/stv-master
+    Hostname ${stv-dev-worker-1-ip:?}
+    ProxyCommand ssh -W %h:%p stv-dev-gateway
+
+Host stv-dev-worker-2
+    User fedora
+    IdentityFile ~/.ssh/stv-master
+    Hostname ${stv-dev-worker-2-ip:?}
+    ProxyCommand ssh -W %h:%p stv-dev-gateway
+
+
+...  
+
+Host stv-dev-zeppelin
+    User fedora
+    IdentityFile ~/.ssh/stv-master
+    Hostname ${stv-dev-zeppelin-ip:?}
+    ProxyCommand ssh -W %h:%p stv-dev-gateway
+
+..
+
+
+
+#-------------------------------------------------------------------------
+## Step 5: Install HDFS Cluster
+#-------------------------------------------------------------------------
+
+## Fetch Hadoop Binaries on each node & install java
+
+
+## Run on all nodes
```

Comment:
What location is the tar.gz unpacked ?

Reply:
tar.gz was unpacked under the fedora user's home directory  (/home/fedora). Notes updated to make this clearer.